lib/services/local/users.go

```diff
@@ -1239,6 +1288,54 @@ func (s *IdentityService) UpsertMFADevice(ctx context.Context, user string, d *t
 	return nil
 }
 
+// upsertUserStatusMFADevice updates the user's MFA state based on the devices
+// they have.
+// It's called after adding or removing an MFA device and ensures the user's
+// MFA state is up-to-date and reflects the weakest MFA device they have.
+func (s *IdentityService) upsertUserStatusMFADevice(ctx context.Context, user string) error {
+	devs, err := s.GetMFADevices(ctx, user, false)
+	if err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
Do we need to read all user devices if we are building the state gradually? If the added device is already the weakest, or the known state it the weakest, then we don't need to update.

In a different approach, should we calculate this during user reads instead of device updates?

Reply:
We can't. The goal is to have access to when a user is updated as well so it can be streamed to access graph.
We can only compute it during reads when `withSecrets=true`, otherwise this information isn't available. 

When using a watcher, user's updates to secrets are ignored (withSecrets=true|false have the same output)

Reply:
We could still early exit in various cases, if we considered the device being added and the current known state, right?

Reply:
Yes, the issue is that the majority of users don't have this field set once we roll out the feature so we will need to read all devices when updating. Most updates happen when a user adds an MFA device on his own so it's always better to reconcile with all scenarios.

Reply:
On rollout there will be a lot of backfilling, but after that I suspect this would be rather stable. Still, mfa registration don't happen that often, so I yield.